legion/engine/physics/data/physics_component_data.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#include <physics/data/colliders/collider_data.hpp>
+#include <physics/events/events.hpp>
+#include <physics/data/component_flags.hpp>
+#include <bitset>
+
+namespace legion::physics
+{
+    constexpr float boxExtentSizeMultiplier = 0.5f;
+
+    class PhysicsComponentData
+    {
+    public:
+
+        inline void AddBoxCollider(const math::vec3& extents, const math::vec3& offset, const math::quat& rotation)
+        {
+            m_convexColliderData.push_back(ConvexColliderData(offset,rotation,extents * boxExtentSizeMultiplier,&m_modificationFlags));
```

Comment:
using a pointer to data in a component might not always stay valid. when the component family needs to grow it's container it reallocates and moves it's items to the new storage. thus this pointer will then be pointing to garbage data.

one way to work around this is to have a `std::unique_ptr<physics_component_data>` inside physics_component instead

Reply:
the idea with passing the bitset to the collider is so that the bitset can be modified when we modify a collider (changing its extents, changing its mesh, etc). I guess what we can do instead is when a user call get collider they instead get a handle that contains a pointer to that collider and a ptr to that bitset

Reply:
another option is to have the getters in the collider classes but the setters in the PhysicsComponentData class

Reply:
well if the component doesn't store the physics_component_data, but just a pointer, and the actual physics_component_data is dynamically allocated. then it doesn't matter if the component moves. any references to physics_component_data are still valid

Reply:
fair enough, although tbh i feel like you'll already have quite a lot of pointer indirection. i'm not so sure whether either will rly be faster than the other without just straight up testing it